cairo/tests/ethereum/cancun/test_state.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+from ethereum_types.bytes import Bytes32
+from hypothesis import given
+
+from ethereum.cancun.fork_types import Address
+from ethereum.cancun.state import State, get_account, get_account_optional, get_storage
+from tests.utils.strategies import address_strategy, key_strategy, state
+
+pytestmark = pytest.mark.python_vm
+
+
+class TestState:
+    @given(state=state, address=address_strategy(state))
```

Comment:
Is this working ?
You want to use the same state in the address_strategy right ?
Instead, shouldn't this be a single composite strategy ?
https://hypothesis.readthedocs.io/en/latest/data.html#chaining-strategies-together

Other solution could be to access address in the state directly instead of having an address parameter

Reply:
> Other solution could be to access address in the state directly instead of having an address parameter

I want to be able to query an address that isn't in the state

Reply:
> Instead, shouldn't this be a single composite strategy ?

how?

Reply:
Maybe something like
```
@given(state=state, take_address_from_state=st.bool)
def test_get_account(...):
       address =  state.address[0] if take_address_from_state else draw(address_strategy)
       
```